### DISCUSSION TITLE
So If I just buy a certificate, download it and put it on the server it will work?

### DISCUSSION THREAD

**Discussion by Frtrillo** [emqx/emqx#8134] [Q&A] [answered]
So I built my emqx cluster with 2 servers and a load balancer, but wss doesnt work on any browser, it only works throught ws.

I've been following this tutorial: https://www.emqx.com/en/blog/emqx-server-ssl-tls-secure-connection-configuration-guide
for self signed but I cant use self signed for every device... I want the usual ssl working, and my load balancer doesnt have tsl termination for wss, only for https (so it doesnt work either).

So If I just buy a ssl certificate like the tutorial says and set it up on the server and configure these 2 lines at emqx.cnf, it will work WSS connections?

**Comment by zmstone** [comment, marked as answer]
Hi @Frtrillo 
Yes. Self-signed is only for demo in the article.
Purchased should work the same way.

**Reply by Frtrillo** [reply]
Hi, thanks for you answer.
But I've been reading and wont't I need to generate a CSR code on the emqx server for the purchase certificate ?
The tutorial only says to "Copy the renamed file mentioned earlier emqx.key and emqx.crt into the directory etc/certs/ of EMQX, and refer the following configuration to modify emqx.conf"

**Reply by zmstone** [reply]
Hi
I have no experience purchasing certs myself.
So I am not entirely sure about the following, but:

I believe CSR can be generated anywhere, it’s the host(or domain) name matters in the CSR.
E.g. If you have the private key in your laptop, you can generate a CSR from you laptop for a server to use.
